src/fortran_utils.F90

```diff
@@ -39,5 +39,30 @@ function int_to_txt_pad(val, width) result(txt)
     allocate(character(w) :: txt)
     write(txt,'(i0.'//widthtxt//')') val
   end function
+  
+  
+  ! using EXECUTE_COMMAND_LINE to call mkdir sometimes fail (EXECUTE_COMMAND_LINE is forked as an new process, which may be the problem)
```

Comment:
Hi Jan, why do you write the C code for that? We generally use system command to avoid all these troubles which costs just 3 lines of code.
As an example:
    !    f%dump_dir='DUMP/'
    !    INQUIRE(file=trim(f%dump_dir), EXIST=f%L_EXISTS)
    !    if (.not. f%L_EXISTS) call system('mkdir '//trim(f%dump_dir))

Reply:
Hi Dima, I did not really write any C code, I am just calling it from Fortran.
EXECUTE_COMMAND_LINE (or the older system) sometimes does not work and it is really hard to find out why. Fortran does it in a complicated way and has to fork a subprocess. My guess is that there are sometimes problems to do a fork due to memory problems or file count limitations. JSC support approves with the idea.
@koldunovn recently had this problem too, see https://github.com/FESOM/fesom2/issues/264#issuecomment-1015952096

Reply:
@hegish But what happens if you just use `system`? It seem to just work fine in other parts of the code.

Reply:
We also have EXECUTE_COMMAND_LINE in other parts of the code. I have also seen the error SYSTEM instead EXECUTE_COMMAND_LINE on aleph (I had them both active for a while). SYSTEM seems to behave similar to EXECUTE_COMMAND_LINE. The problem is, the error happens only rarely, see our discussion at https://github.com/FESOM/fesom2/issues/264.
Currently I have no better idea how to solve this.

Reply:
I see. It's sad that one should have so much code for something that should be essentially a one liner :)